src/DependencyInjection/Configuration.php

```diff
@@ -58,7 +58,15 @@ public function getConfigTreeBuilder()
                                     ->thenInvalid('Invalid consistency value "%s"')
                                 ->end()
                             ->end()
-                            ->integerNode('default_pagesize')->deFaultValue(10000)->end()
+                            ->scalarNode('default_pagesize')
+                                ->deFaultValue(10000)
+                                ->validate()
+                                    ->ifTrue(function ($v) {
+                                        return !is_null($v) && !is_int($v);
```

Comment:
Not sure, but :

php > $v = -1;
php > var_dump(!is_null($v) && !is_int($v));
bool(false)

php > $v = 1;
php > var_dump(!is_null($v) && !is_int($v));                                                                                  
bool(false)

What do you think of :

->ifFalse(function ($v) {
    return is_null($v) || (is_int($v) && $v > 0);
}


php > $v = -1;
php > var_dump( is_null($v) || (is_int($v) && $v > 0) );
bool(false)                                                                                                                                                                                                                                                                                                                  
php > $v = 1;
php > var_dump( is_null($v) || (is_int($v) && $v > 0) );
bool(true)

php > $v = null;
php > var_dump( is_null($v) || (is_int($v) && $v > 0) );
bool(true)